### ISSUE TITLE
Add "stackwalk_version" to the JSON schema?

### ISSUE THREAD

**Issue by Gankra** [luser/rust-minidump#368]
socc-pair keeps pulling this value down now. I'm not sure if this was something @willkg was working on and didn't finish, or something he hacks into the JSON as a post-processing step, but it's not something we actually produce.

I do *like* the idea of the JSON containing some versioning info, as this tends to be helpful for maintaining a stable schema that users can manage updates to. We also currently always publish all the crates in this repo with lockstep versions, so minidump-processor and minidump-stackwalk *do* always agree on the version.

That said, I would like us to have some ability to distinguish between "published" releases and prereleases (random commits). Something like "stackwalk_version": "0.9.8-alpha", so that there's some hint that the user has a non-standard copy and the semantics of the format are a blur between two versions (possibly with transient breaks!).

**Comment by Gankra** [comment]
If there's some macro or whatever to get the version from the Cargo.toml, perhaps the "post release" commit should update everything to NEXT_VERSION-alpha.

e.g. the tree looks like:

* some_random_commit: "minidump-stackwalk 1.0.2-alpha"
* 1.0.2 release: "minidump-stackwalk 1.0.2"
* post release cleanup commit: "minidump-stackwalk 1.0.3-alpha"

I would need to experiment with this a bit.

**Comment by luser** [comment]
> If there's some macro or whatever to get the version from the Cargo.toml, perhaps the "post release" commit should update everything to NEXT_VERSION-alpha.

You're already using one :slightly_smiling_face: :
https://github.com/luser/rust-minidump/blob/d8b246d6a75348c5341859cf5edfa4148a6a973a/minidump-stackwalk/src/main.rs#L27

**Comment by luser** [comment]
I like [`cargo-release`](https://github.com/crate-ci/cargo-release) for automating the whole version bump-tag-commit-version bump + `cargo publish` cycle. I don't think I've tried it with a workspace like this, so it's worth verifying that it works OK, but I've found it to be a huge time saver for other crates like `sccache`:
https://github.com/mozilla/sccache/blob/18614d55aa7b646b5d3ab323c5f68b29b1f5a12b/Cargo.toml#L3
https://github.com/mozilla/sccache/commit/6b6d2f7d2dceefeb4f583712aa4c221db62be0bd#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542

If you suspect you're going to wind up using arbitrary versions from `master` in prod (this feels inevitable), you might also want to stick the git rev in there for pre-release versions. (IDK where the binaries that Socorro uses are being built? You could just pass the rev in an env var if desired.)

**Comment by Gankra** [comment]
Yeah we're already using arbitrary version in prod. Will appends some metadata to get more detailed versions strings in socorro like 

`minidump-stackwalk 0.9.6 aef123eaef23 (2021-11-12)`

**Comment by willkg** [comment]
Socorro doesn't depend on the format of that string at all. It's structured that way because I figured we'd have questions like:

1. which specific version did this processing?
2. oh, but we don't pin to _versions_, we pin to commits, so which commit did this processing?
3. how old is that commit?

I'm game for whatever you want to populate it with. Plus Socorro can always augment that with additional details if we want it to do that.

Also, I stuck the data in the `json_dump` along with the rest of the stackwalker output because Socorro runs minidump-stackwalk on all the minidumps and this was the best way to keep the metadata with the data it was meta-ifying. That's not set in stone--we can change it.

I have another bug for rethinking the schema for a "processed crash" that better splits out metadata around _how_ it was ingested from actual data so it's clearer what came from where and it's easier to work with. I think working on that is a bit further out, but these sorts of problems are all related.

**Comment by luser** [comment]
It's an excellent thing to have because you definitely run into situations where something looks broken but you're pretty sure it was fixed at some point, so you can verify whether the version with the fix was in use.

**Comment by willkg** [comment]
The super duper magic is that it's indexed in Elasticsearch so we can reprocess all the crash reports that were processed with version XYZ which takes so much risk out of minidump-stackwalk issues.
